appmap/_implementation/testing_framework.py

```diff
@@ -0,0 +1,123 @@
+import inflection
```

Comment:
Did you have to make any changes to this code after you copied it out of pytest.py?

Reply:
`FuncItem` was mostly unmodified, except I changed it so it takes generic `cls`, `name` and `method_id` arguments instead of the pytest-specific item. I've simplified the session setup code a bit (using `pathlib` instead of `os`, and generalizing it) and especially the recording part -- I changed it to be a context manager, and instead of using `exit_hook` I simply write the appmap after `yield` continuation. I've found the hook approach a bit confusing and unnecessary.